components/script/dom/teereadrequest.rs

```diff
@@ -0,0 +1,227 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use std::cell::Cell;
+use std::rc::Rc;
+
+use dom_struct::dom_struct;
+use js::jsapi::Heap;
+use js::jsval::{JSVal, UndefinedValue};
+use js::rust::HandleValue as SafeHandleValue;
+
+use super::bindings::structuredclone;
+use super::types::TeeUnderlyingSource;
+use crate::dom::bindings::reflector::{DomObject, Reflector};
+use crate::dom::bindings::root::Dom;
+use crate::dom::bindings::trace::RootedTraceableBox;
+use crate::dom::globalscope::GlobalScope;
+use crate::dom::promise::Promise;
+use crate::dom::readablestream::ReadableStream;
+use crate::microtask::Microtask;
+use crate::script_runtime::CanGc;
+
+#[derive(JSTraceable, MallocSizeOf)]
+#[allow(crown::unrooted_must_root)]
+pub struct TeeReadRequestMicrotask {
+    #[ignore_malloc_size_of = "mozjs"]
+    chunk: Box<Heap<JSVal>>,
+    tee_read_request: Dom<TeeReadRequest>,
+}
+
+impl TeeReadRequestMicrotask {
+    pub fn microtask_chunk_steps(&self, can_gc: CanGc) {
+        self.tee_read_request.chunk_steps(&self.chunk, can_gc)
+    }
+}
+
+#[dom_struct]
+pub struct TeeReadRequest {
```

Comment:
Same here, we can document it with https://streams.spec.whatwg.org/#ref-for-read-request%E2%91%A2, and we will have to later rename it to differentiate it from https://streams.spec.whatwg.org/#ref-for-read-request%E2%91%A3